README: update define class content (#133)
fauna/fauna-jvm: README.md

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutionException;

import com.fauna.client.Fauna;
import com.fauna.client.FaunaClient;
import com.fauna.exception.FaunaException;
import com.fauna.query.builder.Query;
import com.fauna.response.QuerySuccess;
import com.fauna.types.Page;

> import static com.fauna.codec.generic.pageOf;

Reviewer: This doesn't work for me.
```suggestion
import com.fauna.codec.PageOf;
```

Author: oops. it should be `import static com.fauna.codec.Generic.pageOf`;

Reviewer: Nice. I updated my suggestion.

        // Use `query()` to run a synchronous query.
        // Synchronous queries block the current thread until the query completes.
        // Accepts the query, expected result class, and a nullable set of query options.

>         QuerySuccess<Page<Product>> result = client.query(query, pageOf(Product.class));

Reviewer: When I tested, I still needed to use `new PageOf<>`. I might be missing something tho.

Author: hmmm, there's a static import that should work. it's just a helper method

Reviewer: :doh: Yeah — Fixing the import fixed this. Please ignore.

        printResults(result.getData());
    }

    private static void runAsynchronousQuery(FaunaClient client, Query query) throws ExecutionException, InterruptedException {
        // Use `asyncQuery()` to run an asynchronous, non-blocking query.
        // Accepts the query, expected result class, and a nullable set of query options.

>         CompletableFuture<QuerySuccess<Page<Product>>> futureResult = client.asyncQuery(query, pageOf(Product.class));

Reviewer: Same as above.

Reviewer: Please ignore.

options](#query-options).

> ### Define a custom class for your data
> These should be simple POJOs.

Reviewer: ```suggestion
### Define a custom class for your data

Use annotations to map a Java class to a Fauna document or object shape:
```

```java
import com.fauna.annotation.FaunaField;
import com.fauna.annotation.FaunaId;

class Person {
    
    @FaunaId
    private String id;
    
    private String firstName;

    @FaunaField( name = "dob")
    private String dateOfBirth;
}
```

> You can use the `com.fauna.annotation` package to modify encoding and decoding of 

Reviewer: ```suggestion
Use the `com.fauna.annotation` package to modify encoding and decoding of 
```

specific fields in classes used as arguments and results of queries.

> * `@FaunaId`: Should only be used once per class and be associated with a field named `id` if it represents the ID of a document. It will not be encoded unless the `isClientGenerated` flag is set.

Reviewer: ```suggestion
* `@FaunaId`: Should only be used once per class and be associated with a field named `id` that represents the Fauna document ID. It's not encoded unless the `isClientGenerated` flag is`true`.
```

> * `@FaunaTs`: Should only be used once per class and be associated with a field named `ts` if it represents the timestamp of a document. It will never be encoded.

Reviewer: ```suggestion
* `@FaunaTs`: Should only be used once per class and be associated with a field named `ts` that represents the timestamp of a document. It's not encoded.
```

> * `@FaunaColl`: Typically goes unmodeled. Should only be used once per class and be associated with a field named `coll` if represents the collection field of a document. It will never be encoded.

Reviewer: ```suggestion
* `@FaunaColl`: Should only be used once per class and be associated with a field named `coll` that represents the collection of a document. It's not encoded and is not typically modeled. 
```

* `@FaunaField`: Can be associated with any field to override its name in Fauna.
* `@FaunaIgnore`: Can be used to ignore fields during encoding and decoding.

> In the `com.fauna.codec` package, you use classes to handle type erasure when the top-level result
> of a query is a generic, including:

Reviewer: ```suggestion
Use classes in the `com.fauna.codec` package to handle type erasure when the top-level result
of a query is a generic, including:
```

* `PageOf<T>` where `T` is the element type.
* `ListOf<T>` where `T` is the element type.
* `MapOf<T>` where `T` is the value type.
* `OptionalOf<T>` where `T` is the value type.

> * `NullableOf<T>` where `T` is the value type. This is specifically for cases when returning a Fauna Document that may be null and you want to receive a concrete NullDoc<T> or NonNullDoc<T> instead of catching a NullDocumentException.

Reviewer: ```suggestion
* `NullableOf<T>` where `T` is the value type. This is specifically for cases when you return a Fauna document that may be null and want to receive a concrete `NullDoc<T>` or `NonNullDoc<T>` instead of catching a `NullDocumentException`.
```
